### DISCUSSION TITLE
How to clear all caches at once.

### DISCUSSION THREAD

**Discussion by minht11** [posva/pinia-colada#56] [Questions] [answered]
At logout I want to clear all current query cache. With tanstack you can use `queryClient.clear()`. How can I do that?

Also what is is the differences between `query.refetch` and `query.refresh`

**Comment by posva** [comment, marked as answer]
- You can clear all entries with

  ```ts
    const caches = useQueryCache() // must be used within a setup like any composable
    for (const entry of caches.getEntries()) {
      caches.remove(entry)
    }
  ```

- refetch ignores fresh status while refresh doesn’t refetch fresh data
